src/features/contact/contactverzoek/overzicht/service.ts

```diff
@@ -15,69 +17,104 @@ import {
-  url.searchParams.set("ordering", "-record__data__registratiedatum"); //todo: is dit de correcte orderning?
-  url.searchParams.set("pageSize", "10");
-
-  url.searchParams.set(
-    "data_attrs",
-    `betrokkene__digitaleAdressen__icontains__${query}`,
-  ); //todo: kan je in 1 call op email OF telefoonnr zoeken? anders de interface aanpassen zodat duidelijk is dat je maar naar een van beide kan zoeken (bv samenvoegen in 1 zoekveld!)
+  let url: URL;
+
+  if (gebruikKlantInteractiesApi.value) {
+    url = new URL(klantinteractiesDigitaleAdressen, location.origin);
+    url.searchParams.set("adres", query);
+    url.searchParams.set("expand", "verstrektDoorBetrokkene");
+    url.searchParams.set("page", "1");
+  } else {
+    url = new URL("/api/internetaak/api/v2/objects", location.origin);
+    url.searchParams.set("ordering", "-record__data__registratiedatum");
+    url.searchParams.set("pageSize", "10");
+    url.searchParams.set(
+      "data_attrs",
+      `betrokkene__digitaleAdressen__icontains__${query}`
+    );
+  }
 
   return url.toString();
 }
 
 function searchRecursive(urlStr: string, page = 1): Promise<any[]> {
-  //recursive alle pagina's ophalen.
-  //set de page param van de huidige op te halen pagina
   const url = new URL(urlStr);
   url.searchParams.set("page", page.toString());
 
-  return (
-    fetchLoggedIn(url)
-      .then(throwIfNotOk)
-      .then(parseJson)
-      //todo: parsePagination toevoegen??
-      .then(async (j) => {
-        if (!Array.isArray(j?.results)) {
-          throw new Error("expected array: " + JSON.stringify(j));
-        }
-
-        const result: any[] = [];
-        j.results.forEach((k: any) => {
-          result.push(k);
-        });
-
-        if (!j.next) return result;
-        return await searchRecursive(urlStr, page + 1).then((next) => [
-          ...result,
-          ...next,
-        ]);
-      })
-  );
+  return fetchLoggedIn(url)
+    .then(throwIfNotOk)
+    .then(parseJson)
+    .then(async (j) => {
+      if (!Array.isArray(j?.results)) {
+        throw new Error("Expected array: " + JSON.stringify(j));
+      }
+
+      const result: any[] = [...j.results];
+
+      if (!j.next) return result; 
+      const nextResults = await searchRecursive(urlStr, page + 1);
+      return [...result, ...nextResults];
+    });
 }
 
-function search(url: string) {
-  return searchRecursive(url); //todo: sortering???  .then((r) => r.sort(sortKlant));
+export async function search(
+  query: string,
+  gebruikKlantInteractiesApi: Ref<boolean | null>
+): Promise<PaginatedResult<Contactverzoek>[]> { 
+  const url = getSearchUrl(query, gebruikKlantInteractiesApi);
+
+  if (gebruikKlantInteractiesApi.value) {
+    const searchResults = await searchRecursive(url);
+    const enrichedResults = await Promise.all(
+      searchResults.map(async (result: any) => {
+        const klantUrl = ref(result.url);
+        return await getContactverzoekenByKlantUrl(klantUrl);
```

Comment:
Dit snap ik niet. Je zoekt op digitale adressen, toch? hoe kan result.url dan een klantUrl zijn? dat is dan toch een digitaalAdresUrl?

Reply:
ik bedoelde denk url van openklant. is aangepast